app/jobs/regular/unapprove_past_category_expert_posts.rb

```diff
@@ -16,25 +14,46 @@ def execute(args)
         raise Discourse::InvalidParameters.new(:category_ids)
       end
 
-      posts = Post.joins(:topic)
-        .where(user_id: args[:user_id])
-        .where(topic: { category_id: args[:category_ids] })
+      user = User.find_by(id: args[:user_id])
+      raise Discourse::InvalidParameters.new(:user_id) unless user
+
+      # The user was removed as category expert from 1 group. They could
+      # still be a category expert by another group memebership.
+      # Filter out those categories.
+      category_ids = args[:category_ids].select do |category_id|
```

Comment:
This ends up doing one query per category_id. I'd recommend something like this instead:

```ruby
Category.where(id: args[:category_ids]).filter do |c|
  user.expert_group_ids_for_category(c).empty?
end
```

Reply:
@eviltrout Much much better, thanks. Committed